packages/react/src/components/TextField/TextField.test.tsx

```diff
@@ -731,81 +695,79 @@ describe('TextField onChange', () => {
 
-    const input = wrapper!.find('input');
-    // Fire input AND change events to more realistically test
-    input.simulate('input', mockEvent(changeValue));
-    input.simulate('change', mockEvent(changeValue));
+    const input = getInput();
+    fireEvent.change(input, { target: { value: changeValue } });
 
     expect(onChange).toHaveBeenCalledTimes(calls);
     expect(textField!.value).toEqual(expectedValue);
-    expect((input.getDOMNode() as HTMLInputElement).value).toEqual(expectedValue);
+    expect(input.value).toEqual(expectedValue);
   }
 
   beforeEach(() => {
     onChange = jest.fn();
   });
 
   it('should be called for input change and apply edits if uncontrolled', () => {
-    wrapper = mount(<TextField componentRef={textFieldRef} defaultValue={initialValue} onChange={onChange} />);
+    render(<TextField componentRef={textFieldRef} defaultValue={initialValue} onChange={onChange} />);
 
     expect(onChange).toHaveBeenCalledTimes(0);
     simulateAndVerifyChange('value change', 1);
     simulateAndVerifyChange('', 2);
   });
 
   it('should not be called when initial value is undefined and input change is an empty string', () => {
-    wrapper = mount(<TextField componentRef={textFieldRef} onChange={onChange} />);
+    render(<TextField componentRef={textFieldRef} onChange={onChange} />);
 
     expect(onChange).toHaveBeenCalledTimes(0);
     simulateAndVerifyChange('', 0);
   });
 
   it('should apply edits if implicitly uncontrolled', () => {
-    wrapper = mount(<TextField componentRef={textFieldRef} onChange={onChange} />);
+    render(<TextField componentRef={textFieldRef} onChange={onChange} />);
 
     simulateAndVerifyChange('value change', 1);
   });
 
   it('should not apply edits if controlled', () => {
-    wrapper = mount(<TextField componentRef={textFieldRef} value={initialValue} onChange={onChange} />);
+    render(<TextField componentRef={textFieldRef} value={initialValue} onChange={onChange} />);
 
     expect(onChange).toHaveBeenCalledTimes(0);
     simulateAndVerifyChange('value change', 1, initialValue);
   });
 
   it('should not apply edits if controlled (empty initial value)', () => {
-    wrapper = mount(<TextField componentRef={textFieldRef} value="" onChange={onChange} />);
+    render(<TextField componentRef={textFieldRef} value="" onChange={onChange} />);
 
     expect(onChange).toHaveBeenCalledTimes(0);
     simulateAndVerifyChange('value change', 1, '');
   });
 
   it('respects prop updates in response to onChange', () => {
     onChange = jest.fn((ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, value?: string) =>
-      wrapper!.setProps({ value }),
+      rerender(<TextField componentRef={textFieldRef} value={value} onChange={onChange} />),
     );
-    wrapper = mount(<TextField componentRef={textFieldRef} value="" onChange={onChange} />);
+    const { rerender } = render(<TextField componentRef={textFieldRef} value="" onChange={onChange} />);
```

Comment:
I don't think rerender is being used here?
```suggestion
    render(<TextField componentRef={textFieldRef} value="" onChange={onChange} />);
```

Reply:
it's used in the `onChange` function above :)